extensions-contrib/opentsdb-emitter/src/main/java/org/apache/druid/emitter/opentsdb/EventConverter.java

```diff
@@ -41,17 +41,24 @@
   private static final Pattern WHITESPACE = Pattern.compile("[\\s]+");
 
   private final Map<String, Set<String>> metricMap;
+  private final String namespacePrefix;
 
-  public EventConverter(ObjectMapper mapper, String metricMapPath)
+  public EventConverter(ObjectMapper mapper, String metricMapPath, String namespacePrefix)
   {
     metricMap = readMap(mapper, metricMapPath);
+    this.namespacePrefix = namespacePrefix;
   }
 
   protected String sanitize(String metric)
   {
     return WHITESPACE.matcher(metric.trim()).replaceAll("_").replace('/', '.');
   }
 
+  private String buildNamespace()
+  {
+    return (namespacePrefix == null || "".equals(namespacePrefix)) ? "" : sanitize(namespacePrefix) + ".";
```

Comment:
can `namespacePrefix` actually be empty here (other than the unit tests creating an `EventConverter` directly) since `OpentsdbEmitterConfig` looks like it will coerce the value to `null` in its constructor?

Reply:
You're right, i've removed the tests for it, since it's the only place where it can happen. And i guess no-one is gonna call `EventConverter` outside of the extension.